# HG changeset patch
# User Rik <rdrider0-list@yahoo.com>
# Date 1261085309 28800
#      Thu Dec 17 13:28:29 2009 -0800
# Node ID 8561ecc4431701e0fc6507dc8163b504f133c598
# Parent  929e53107f09c413a9c3a4d1f99d11be4ba6ae4d
Use automake syntax to build munge-texi

diff --git a/doc/ChangeLog b/doc/ChangeLog
--- a/doc/ChangeLog
+++ b/doc/ChangeLog
@@ -1,8 +1,12 @@
+2009-12-17  Rik  <octave@nomad.inbox5.com>
+
+	* interpreter/Makefile.am: use automake syntax to build munge-texi.
+
 2009-12-11  Rik  <octave@nomad.inbox5.com>
 
 	* interpreter/Makefile.am: remove install-local target and use automake 
 	syntax to install doc-cache file.
 
 2009-12-10  Rik <octave@nomad.inbox5.com>
 
 	* interpreter/Makefile.am: Replace distdir-local with dist-hook rule as it
diff --git a/doc/interpreter/Makefile.am b/doc/interpreter/Makefile.am
--- a/doc/interpreter/Makefile.am
+++ b/doc/interpreter/Makefile.am
@@ -28,16 +28,20 @@ TEXINFO_TEX = ../texinfo.tex
 ## Leading PATH_SEPARATOR required due to weak parsing by dvips (12/04/09)
 TEXINPUTS := "$(PATH_SEPARATOR)$(srcdir)$(PATH_SEPARATOR)$(TEXINPUTS)$(PATH_SEPARATOR)"
 export TEXINPUTS
 
 ## Include custom texmf.cnf necessary to run @seealso macro 
 TEXMFCNF := "..$(PATH_SEPARATOR)$(srcdir)/..$(PATH_SEPARATOR)$(TEXMFCNF)$(PATH_SEPARATOR)"
 export TEXMFCNF
 
+## Automake generated rules for documentation are not parallel-safe.
+## Restrict current directory to run serially
+.NOTPARALLEL:
+
 dist_man1_MANS = \
   mkoctfile.1 \
   octave-bug.1 \
   octave-config.1 \
   octave.1
 
 ## The following example files are listed for dependencies.
 ## They should not be distributed from this directory.
@@ -166,20 +170,22 @@ dist-hook:
 
 octetc_DATA = doc-cache
 
 DOCSTRING_FILES = $(shell $(srcdir)/find-docstring-files.sh "$(top_srcdir)")
 
 doc-cache: $(DOCSTRING_FILES) mk_doc_cache.m
 	$(TOPDIR)/run-octave -f -q -H $(srcdir)/mk_doc_cache.m doc-cache $(DOCSTRING_FILES) || { rm -f doc-cache; exit 1; }
 
-$(MUNGED_TEXI_SRC): $(DOCSTRING_FILES) munge-texi.cc
+## Program compiled only to help build documentation.  No need to install
+noinst_PROGRAMS = munge-texi
 
-munge-texi$(BUILD_EXEEXT): munge-texi.cc
-	$(BUILD_CXX) $(BUILD_CXXFLAGS) -o $@ $^ $(BUILD_LDFLAGS)
+munge_texi_SOURCES = munge-texi.cc
+
+$(MUNGED_TEXI_SRC): $(DOCSTRING_FILES) $(munge_texi_SOURCES)
 
 .txi.texi:
 	@$(MAKE) $(AM_MAKEFLAGS) munge-texi$(BUILD_EXEEXT)
 	./munge-texi $(top_srcdir) $(DOCSTRING_FILES) < $< > $@-t
 	mv $@-t $@
 
 contributors.texi: contributors.in
 	$(AWK) -f $(srcdir)/mkcontrib.awk $(srcdir)/contributors.in > $@-t
@@ -200,24 +206,20 @@ contributors.texi: contributors.in
 EXTRA_DIST = \
   config-images.sh \
   contributors.in \
   find-docstring-files.sh \
   images \
   images.mk \
   mk_doc_cache.m \
   mkcontrib.awk \
-  munge-texi.cc \
   octave.dvi \
   octave.html \
   octave.pdf \
   octave.ps \
   $(IMAGES) \
   $(IMAGES_SRC) \
   $(TXI_SRC)
 
-DISTCLEANFILES = $(octave_TEXINFOS) doc-cache munge-texi$(BUILD_EXEEXT)
+DISTCLEANFILES = $(octave_TEXINFOS) doc-cache
 
 MAINTAINERCLEANFILES = $(IMAGES)
 
-## Automake generated rules for documentation are not parallel-safe.
-## Restrict current directory to run serially
-.NOTPARALLEL:
